src/services/auth.service.ts

```diff
@@ -0,0 +1,39 @@
+import { type DeepPartial } from 'typeorm'
+import { User } from '../entity/User'
+import AppDataSource from '../data-source'
+import * as bcrypt from 'bcrypt'
+
+class CustomError extends Error {
+  status: number
+  constructor (message: string, status: number) {
+    super(message)
+    this.status = status
+    this.name = this.constructor.name
+  }
+}
```

Comment:
move it to another file : errors/custom-error.ts (for instance)
it might be useful for later case